src/core/client_channel/subchannel.cc

```diff
@@ -53,8 +53,12 @@
 #include "src/core/lib/transport/connectivity_state.h"
 #include "src/core/lib/transport/error_utils.h"
 #include "src/core/lib/transport/transport.h"
+#ifdef GRPC_XDS_ENABLED
```

Comment:
There should not be any xDS-specific code in this file.

For the backend service label, please use [`GRPC_ARG_BACKEND_SERVICE`](https://github.com/grpc/grpc/blob/2f514a8d4b2a5e6c2053767166577f913bd05807/src/core/resolver/endpoint_addresses.h#L46), which is not xDS-specific.

For the locality label, we will need to add a non-xDS-specific way to plumb it.  I suggest adding a new channel arg name in endpoint_addresses.h (right next to where `GRPC_ARG_BACKEND_SERVICE` is defined) called `GRPC_ARG_LB_LOCALITY`.  You can have the cds LB policy set that arg in the [same place where it sets the xDS-specific arg](https://github.com/grpc/grpc/blob/2f514a8d4b2a5e6c2053767166577f913bd05807/src/core/load_balancing/xds/cds.cc#L293).